src/net/mod.rs

```diff
@@ -1093,6 +1093,10 @@ pub struct RPCPoxInfoData {
     pub reward_cycle_length: u64,
     pub rejection_votes_left_required: u64,
     pub next_reward_cycle_in: u64,
+
+    // Stacks 2.1 / PoX-2 info
+    pub pox_2_activation_burnchain_block_height: u64,
```

Comment:
Can you organize these into a separate sub-object that describes each version of PoX, such as the following?

```
"contract_versions": [
   {
     "contract_id": "SP000000000000000000002Q6VF78.pox",
     "activation_burnchain_block_height": 666050,
     "first_reward_cycle_id": 0
   },
   {
     "contract_id": "SP000000000000000000002Q6VF78.pox-2",
     "activation_burnchain_block_height": ...
     "first_reward_cycle_id": ...
   }
]
```

This would future-proof this information so if there's ever a `pox-3` or subsequent revisions, this field's structure would remain semantically unchanged (it's just that the array of contract version objects would grow by one for each additional revision).

Reply:
You'd also want to add a test for this, such as by modifying `test_rpc_getpoxinfo()`.

Reply:
Great idea, done except for the test, I'm not quite sure what to change for that.